spopt/locate/util.py

```diff
@@ -30,40 +36,65 @@ def simulated_geo_points(in_data, needed, seed) -> geopandas.GeoDataFrame:
     >>> street = spaghetti.element_as_gdf(ntw, arcs=True)
 
     >>> street_buffered = geopandas.GeoDataFrame(
-    ...                            geopandas.GeoSeries(street["geometry"].buffer(0.3).unary_union),
-    ...                            crs=street.crs,
-    ...                            columns=["geometry"])
+    ...     geopandas.GeoSeries(street["geometry"].buffer(0.3).unary_union),
+    ...     crs=street.crs,
+    ...     columns=["geometry"]
+    ... )
 
     >>> points_simulated = simulated_geo_points(street_buffered, needed=10, seed=1)
     >>> type(points_simulated)
     <class 'geopandas.geodataframe.GeoDataFrame'>
 
     """
 
-    geoms = in_data.geometry
-    area = tuple(
-        in_data.total_bounds
-    )  # create a polygon with bounds to represent an area
+    # ensure ``needed`` is a positive whole number
+    if isinstance(needed, float) or needed < 1:
+        msg = (
+            f"Cannot generate {needed} points. "
+            "``needed`` must be a positive whole number."
+        )
+        if needed < 1:
+            raise ValueError(msg)
+        if not (needed % int(needed)):
+            needed = int(needed)
+        else:
```

Comment:
For example, if a user passes in ``needed=5.0``. Since ``int(5.0)==5`` we can easily cast to `int`, but the float must be whole number. So here we can check the remainder of ``float(needed)/int(needed)``. If no remainder, we are safe to continue. If there is a remainder, raise an exception.

```python
>>> 5.0 % 5 # safe to cast as int
0
>>> 5.5 % 5 # unsafe to cast as int
0.5
```